assets/src/edit-story/components/panels/animation/directionRadioInput.js

```diff
@@ -0,0 +1,204 @@
+        case DIRECTION.LEFT_TO_RIGHT:
+          return '90deg';
+        default:
+          return '0deg';
+      }
+    }}
+  );
+`;
+
+const Icon = styled.div`
+  padding: 4px;
+`;
+
+const Direction = ({ className, direction }) => (
+  <Icon className={className}>
+    <Svg viewBox="0 0 10 11" direction={direction}>
+      <path d="M5 11L5 1M5 1L9 5M5 1L1 5" />
+    </Svg>
+  </Icon>
+);
+
+Direction.propTypes = {
+  className: PropTypes.string,
+  direction: PropTypes.oneOf(Object.values(DIRECTION)),
+};
+
+// Must be a styled component to add component selectors in css
+const DirectionIndicator = styled(Direction)``;
+
+const Fieldset = styled.fieldset`
+  position: relative;
+  height: 63px;
+  width: 63px;
+  background-color: ${({ theme }) => theme.colors.bg.workspace};
+  border: 1px solid ${({ theme }) => theme.colors.fg.v9};
+  border-radius: 4px;
+`;
+
+const Figure = styled.div`
+  position: absolute;
+  display: block;
+  top: 50%;
+  left: 50%;
+  width: 18px;
+  height: 26px;
+  background: ${({ theme }) => theme.colors.fg.v9};
+  border-radius: 2px;
+  transform: translate(-50%, -50%);
+`;
+
+const Label = styled.label`
+  position: absolute;
+  cursor: pointer;
+
+  ${({ direction }) => {
+    switch (direction) {
+      case DIRECTION.RIGHT_TO_LEFT:
+        return css`
+          top: 50%;
+          right: 0;
+          transform: translateY(-50%);
+        `;
+      case DIRECTION.TOP_TO_BOTTOM:
+        return css`
+          top: 0;
+          left: 50%;
+          transform: translateX(-50%);
+        `;
+      case DIRECTION.LEFT_TO_RIGHT:
+        return css`
+          top: 50%;
+          left: 0;
+          transform: translateY(-50%);
+        `;
+      default:
+        return css`
+          bottom: 0;
+          left: 50%;
+          transform: translateX(-50%);
+        `;
+    }
+  }}
+
+  ${DirectionIndicator} {
+    stroke: ${({ theme }) => theme.colors.fg.v9};
+    stroke-width: 1px;
+  }
+
+  input:checked ~ ${DirectionIndicator} {
+    stroke: ${({ theme }) => theme.colors.activeDirection};
+    stroke-width: 2px;
+  }
+
+  input:focus ~ ${DirectionIndicator} {
+    outline: 5px auto -webkit-focus-ring-color;
+  }
+`;
+
+const hidden = css`
+  position: absolute;
+  opacity: 0;
+  clip-path: polygon(0 0);
+  cursor: pointer;
+`;
+const HiddenLegend = styled.legend`
+  ${hidden}
+`;
+const HiddenCaption = styled.span`
+  ${hidden}
+`;
+const HiddenInput = styled.input`
+  ${hidden}
+`;
+
+const RadioGroup = styled.div`
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+`;
+
+export const DirectionRadioInput = ({
+  directions = [],
+  onChange,
+  defaultChecked,
+}) => {
+  return (
+    <Fieldset>
+      <Figure />
+      <HiddenLegend>{__('Which Direction?', 'web-stories')}</HiddenLegend>
+      <RadioGroup>
+        {directions.map((direction) => (
+          <Label key={direction} htmlFor={direction} direction={direction}>
+            <HiddenCaption>
```

Comment:
Since this span is hidden and only serving to indicate to screen readers what the labeling is, you're better off using aria-label on the `Label` and removing `HiddenCaption`.

Reply:
oooo this is the magic I was looking for. thanks!